controllers/resource/resource_generator.go

```diff
@@ -311,7 +318,56 @@ func createVolumesForDogu(doguResource *k8sv1.Dogu, dogu *core.Dogu) []corev1.Vo
 		volumes = append(volumes, dataVolume)
 	}
 
-	return volumes
+	return volumes, multiError
+}
+
+func createClientVolumeFromDoguVolume(doguVolume core.Volume) (*corev1.Volume, error) {
+	client, clientExists := doguVolume.GetClient(operatorVolumeClientName)
+	if !clientExists {
+		return nil, fmt.Errorf("dogu volume %s has no client", doguVolume.Name)
+	}
+
+	clientParams := new(volumeClientParams)
+	err := convertGenericJsonObject(client.Params, clientParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s client params of volume %s: %w", operatorVolumeClientName, doguVolume.Name, err)
+	}
+
+	switch clientParams.Type {
+	case configMapParamType:
+		configMapParamContent := new(volumeClientConfigMapContent)
+		err = convertGenericJsonObject(clientParams.Content, configMapParamContent)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read %s client type content of volume %s: %w", configMapParamType, doguVolume.Name, err)
+		}
+
+		return &corev1.Volume{
+			Name: doguVolume.Name,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: configMapParamContent.Name},
+				},
+			},
+		}, nil
+	default:
+		return nil, fmt.Errorf("unsupported client param type %s in volume %s", clientParams.Type, doguVolume.Name)
+	}
+}
+
+// convertGenericJsonObject is necessary because go unmarshalls generic json objects as `map[string]interface{}`.
+// Therefore a type assertion is not possible.
+func convertGenericJsonObject(genericObject interface{}, targetObject interface{}) error {
```

Comment:
```suggestion
// convertGenericJsonObject is necessary because go unmarshalls generic json objects as `map[string]interface{}`, 
// and, therefore, a type assertion is not possible. This method marshals the generic object (`map[string]interface{}`) 
// back into a string. This string is then unmarshalled back into a specific given struct. 
func convertGenericJsonObject(genericObject interface{}, targetObject interface{}) error {
```